specs/progress.md

```diff
@@ -0,0 +1,163 @@
+# Progress
+
+## Overview
+
+*Progress* is used to display the length of time a process will take (referred to as a **determinate** state) or to represent an unspecified wait time (referred to as an **indeterminate** state). *Progress* is typically visually represented by a circular or linear animation. When the `value` attribute is passed the state is **determinate**, otherwise it is **indeterminate**.
+
+### Use Cases
+
+The *progress* component should be used for an element that displays the progress status for a task that takes more than a few seconds or consists of several steps. Typical use cases include loading elements or status of a download. 
+
+- Jim is building a dashboard to track fullfilment for his online store he uses progress ring to visualize various metrics.
+
+- Susan is downloading a large collection of research music onto her harddrive. She sees a linear progress displaying the percentage complete for the currently downloading song and indeterminate linear progress for those songs in queue.
+
+- Fin is scrolling a large list of continuously loading data as he scrolls he sees a indeterminate progress ring to let him know that data is loading.
+
+### Types
+*Progress* can be visually represented in 2 ways.
+- Linear - A straight horizontal line. `fast-progress`
+- Circular - A cicular ring shape. `fast-progress-ring`
+
+### Features
+
+A progress should allow the following attributes:
+- `value`, the value of the progress, if not passed the progress will be put into its "indeterminate" state.
+- `min`, the minimum value.
+- `max`, the maximum value.
+- `paused`, whether the progress is paused or not.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/progress)
+- [Material UI](https://material-ui.com/components/progress/)
+- [Lightning Design / Linear](https://www.lightningdesignsystem.com/components/progress-bar/)
+- [Lightning Design / Circular](https://www.lightningdesignsystem.com/components/progress-ring/)
+- [Ant Design](https://ant.design/components/progress/)
+- [Atlassian / Linear](https://atlaskit.atlassian.com/packages/server/progress-bar)
+- [Atlassian / Circular](https://atlaskit.atlassian.com/packages/core/spinner)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/progress-controls)
+
+---
+
+### API
+
+*Component name:*
+- `fast-progress`
+- `fast-progress-ring`
+
+*Attributes:*
+- `value`: number
+- `min`: number
+- `max`: number
+- `paused`: boolean
+
+*Slots:*
+- `indeterminate-linear`
+- `indeterminate-circular`
+
+### Anatomy and Appearance
+
+*Parts:*
+Linear parts:
+- determinate
```

Comment:
The part names seem to be applied more as states than as physical parts of the component. Is there a way we can name these element parts so that we can re-use structure?